autowarefoundation/autoware.universe
# feat(planning_validator): add validation for trajectory length and related tests

## Description

This PR improves planning validation.

- Two validations are added: 
  - check trajectory forward length 
  - check longitudinal deviation
- Add tests for all validation in the ros node level
- Fix some bugs, and missing codes. (These are commented in github)

Now, the planning_validator can detect the invalid planning trajectory more correctly. For example, a very short path can be detected.


**Before**

https://github.com/autowarefoundation/autoware.universe/assets/21360593/914e75d2-db19-481d-abdb-be097e144e87



**After**

https://github.com/autowarefoundation/autoware.universe/assets/21360593/29e89df3-5183-4b0d-ab03-2ddf79bc1c6a


## Related links

None

## Tests performed

- run test in colcon
- run psim and check if undesired behavior occurs (no error message on terminal)

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

The planning validation may will make more errors in diagnostics.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
